src/components/spotlight/SpotlightTypes.tsx

```diff
@@ -0,0 +1,126 @@
+/*
+ * Echoes React
+ * Copyright (C) 2023-2025 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+import { ReactNode } from 'react';
+import {
+  Callback as JoyrideCallback,
+  CallBackProps as SpotlightCallbackProps,
+} from 'react-joyride';
+import { SpotlightModalProps } from './SpotlightModal';
```

Comment:
NP: I'd move the `SpotlightModalProps` in here: I wouldn't expect a `types` module to import from a component module, but rather the opposite.